index.js

```diff
@@ -30,129 +30,128 @@ function processFirstItem(stringList, callback) {
 
 /**
  * ### Challenge `processLength`
- * 
+ *
  * @instructions
  * PLEASE STUDY THE EXAMPLE CHALLENGE THOROUGHLY BEFORE PROCEEDING!
  * PLEASE STUDY THE EXAMPLE CHALLENGE THOROUGHLY BEFORE PROCEEDING!
- * 
+ *
  * Implement a higher-order function called `processLength`.
  * It takes two arguments:
  * @param list an array with elements of any type.
  * @param callback function that takes a number as its argument.
  * @returns the result of invoking `callback` passing the LENGTH of `list`.
- * 
+ *
  * Examples of usage of this higher-order function:
  * [1] Invoking `processLength` passing `['foo', 'bar']` and `(num) => num + 1000`,
  * should return 1002.
- * 
+ *
  * [2] Invoking `processLength` passing `[]` and `(num) => "There are " + num`,
  * should return "There are 0".
 */
-function processLength(/* CODE HERE */) {
-  /* CODE HERE */
+function processLength(list, callback) {
+  return callback(list.length);
 }
 
 /**
  * ### Challenge `processLastItem`
- * 
+ *
  * @instructions
  * Implement a higher-order function called `processLastItem`.
  * It takes two arguments:
  * @param stringList array of strings.
  * @param callback function that takes a string as its argument.
  * @returns the result of invoking `callback` with the LAST element in `stringList`.
- * 
+ *
  * Example of usage of this higher-order function:
  * Invoking `processLastItem` passing `['foo', 'bar']` and `(str) => str + str`,
  * should return 'barbar'.
 */
-function processLastItem(/* CODE HERE */) {
-  /* CODE HERE */
+function processLastItem(stringList, callback) {
+  return callback (stringList.pop());
 }
```

Comment:
Although .pop will give you the last item in the array, it will also alter (or mutate) the array.  We always want to avoid mutating an array unless it's intentional.  Remember that we access the last item in an array by using arr[arr.length-1].